### PR TITLE
Add a progress meter with unspecified number of iterations.

### PR DESCRIPTION
**Pull request by saschatimme**
This addresses #101 from @tpapp.
I am not super happy with the naming, so if somebody has a better idea :)

### CODE REVIEW

**Comment by timholy** [comment]
I like it. Naming is hard. Here are some alternatives I thought of:

- `ProgressIterations`
- `ProgressUnknown`
- `AreWeThereYet`
- `ProgressCounter`

**Comment by timholy** [comment]
Maybe post a screen shot for people who don't want to build this to see what it looks like?

File: src/ProgressMeter.jl [hunk 1]
```diff
@@ -203,6 +237,34 @@ function updateProgress!(p::ProgressThresh; showvalues = Any[], valuecolor = :bl
...
+        msg = @sprintf "%s %d \t Time: %s" p.desc p.counter dur
+        move_cursor_up_while_clearing_lines(p.output, p.numprintedvalues)
+        printover(p.output, msg, p.color)
+        printvalues!(p, showvalues; color = valuecolor)
+        # Compensate for any overhead of printing. This can be
+        # especially important if you're running over a slow network
+        # connection.
+        p.tlast = t + 2*(time()-t)
+        p.printed = true
+    end
```
Comment (timholy) [inline src/ProgressMeter.jl]: Add `return`? Just for return-type stability.

**Comment by saschatimme** [comment]
Here is how it looks: [asciicast](https://asciinema.org/a/SYwEAcPLzmfpA2xmUpeLOoTe4)

**Comment by saschatimme** [comment]
> I like it. Naming is hard. Here are some alternatives I thought of:
> 
> * `ProgressIterations`
> * `ProgressUnknown`
> * `AreWeThereYet`
> * `ProgressCounter`

You mean that you like `ProgressOnline`?

File: src/ProgressMeter.jl (hunk 1 above)
Comment (saschatimme) [reply src/ProgressMeter.jl]: Good catch. Added it.

**Comment by timholy** [comment]
I like the content. I'm not wild about `ProgressOnline`, but it's not obviously worse than the alternatives (though I confess a certain genuine fondness for `AreWeThereYet` :smile:). Anyone else want to chime in?

**Comment by saschatimme** [comment]
I think I like `ProgressUnknown` the most by now. Do we just want to go with that?

**Comment by saschatimme** [comment]
@timholy I would update the PR if you are happy(ish) with `ProgressUnknown` such that we can get this merged.

**Comment by timholy** [comment]
Sure let's go with `ProgressUnknown`. You may not have seen this, but care to respond to https://github.com/timholy/ProgressMeter.jl/issues/101#issuecomment-430488777?

**Comment by timholy** [comment]
Thank you!

**Comment by timholy** [comment]
Just noticed #113. I will tag after taking a chance to review that, too. (Tomorrow, not today.)

**Pull request merged**